### ISSUE TITLE
[Question] Layout Fails for Plain ASCII Text

### ISSUE THREAD

**Issue by cybel** [HOST-Oman/libraqm#99]
So far, I've followed the mini-tutorial in the documentation and hit all the key functions it calls in the same order. The problem seems to be when ``raqm_layout`` is called, it gets to this line: ``if (!rq->text_info[i].ftface)`` (raqm.c:851) and bails. 

As far as I can tell, the font I'm loading up and passing in is completely valid. I'm not sure if I'm doing something wrong or what. Any insight that I can chase would be greatly appreciated.

**Comment by khaledhosny** [comment]
Can you post the relevant code here?

**Comment by cybel** [comment]
Thanks for the incredibly speed in replying. 

Here's a gist with the relevant code https://gist.github.com/cybel/b8b394d0b6b7ccb36e7a07efa89a2e2f

Here's a rundown of how it all works:
``C()`` is the constructor. During that call, ``raqm_set_par_direction`` is called first, the language is set to English. The final call is to ``face(const FT_Font*)`` which sets the font. During this call, ``updateFace()`` and ``updateTexture()`` are called. ``updateFace()`` calls ``FT_Set_Pixel_Sizes`` and ``raqm_set_freetype_face``. Finally, ``updateTexture()`` sets the text ``raqm_set_text_utf8`` and does the layout ``raqm_layout``. 

Hopefully my unusual style won't cause you much grief, but don't hesitate to ask for any other information you need.

**Comment by khaledhosny** [comment]
You need to call `raqm_set_text_utf8()` before `raqm_set_freetype_face()`. That is an undocumented limitation, and I think we should fix it.

**Comment by cybel** [comment]
That solved it! You're a scholar and a gentleman. Thanks!

**Comment by khaledhosny** [comment]
You are welcomed! I’m keeping this issue open as a reminder to fix this limitation.

**Comment by niello** [comment]
You can simply return `false` if `raqm_set_freetype_face*` is called with `rq->text_len` being 0. Otherwise this would require a temporary variable that keeps the value until there is a text to use it. And this behaviour willnot be consistent with `raqm_set_freetype_face_range` because calling this before the text would require to allocate `text_info` before text, messing up the logic.

**Comment by khaledhosny** [comment]
Makes sense. The documentation still needs to make this ordering requirement explicit.
